google/cloud/aiplatform/metadata/schema/base_artifact.py

```diff
@@ -0,0 +1,134 @@
+from typing import Optional, Dict
+
+
+class BaseArtifactSchema(object):
+    """Base class for Metadata Artifact types.
+
+    This is the base class for defining various artifact types, which can be
+    passed to google.Artifact to create a corresponding resource.
+    Artifacts carry a `metadata` field, which is a dictionary for storing
+    metadata related to this artifact. Subclasses from ArtifactType can enforce
+    various structure and field requirements for the metadata field.
+
+    Args:
+        schema_title (str):
+            Optional. The schema title used by the Artifact, defaults to "system.Artifact"
+        resource_name (str):
+            Optional. The resource name of the Artifact following the format as follows.
+            This is globally unique in a metadataStore:
+            projects/123/locations/us-central1/metadataStores/<metadata_store_id>/artifacts/<resource_id>.
+        uri (str):
+            Optional. The uniform resource identifier of the artifact file. May be empty if there is no actual
+            artifact file.
+        display_name (str):
+            Optional. The user-defined name of the Artifact.
+        schema_version (str):
+            Optional. schema_version specifies the version used by the Artifact.
+            If not set, defaults to use the latest version.
+        description (str):
+            Optional. Describes the purpose of the Artifact to be created.
+        metadata (Dict):
+            Optional. Contains the metadata information that will be stored in the Artifact.
+        state (google.cloud.gapic.types.Artifact.State):
+            Optional. The state of this Artifact. This is a
+            property of the Artifact, and does not imply or
+            capture any ongoing process. This property is
+            managed by clients (such as Vertex AI
+            Pipelines), and the system does not prescribe or
+            check the validity of state transitions.
+        **kwargs:
+            Optional. Additional Args that will be passed directly to the Artifact base method for backward compatibility.
+    """
+
+    ARTIFACT_PROPERTY_KEY_RESOURCE_NAME = "resourceName"
+    SCHEMA_TITLE = "system.Artifact"
+
+    def __init__(
+        self,
+        schema_title: Optional[str] = None,
+        resource_name: Optional[str] = None,
+        uri: Optional[str] = None,
+        display_name: Optional[str] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Dict] = None,
+        state: gca_artifact.Artifact.State = gca_artifact.Artifact.State.LIVE,
+        **kwargs,
+    ):
+
+        """Initializes the Artifact with the given name, URI and metadata."""
+        self.schema_title = BaseArtifactSchema.SCHEMA_TITLE
+        if schema_title:
+            self.schema_title = schema_title
+        self.resource_name = resource_name
+
+        self.resource_id = None
+        if resource_name:
```

Comment:
Why do we accept the resource name?

Reply:
I'm sorry I think i miss understood this, I thought the preference was to  accept resouce_name over resouce_id . let me know if we should remove resource name and just accept resouce_id instead.

Reply:
per offline conversation since I'm only using the resource_name to extract the id from  it is better to directly accept the resource_id.  updating the code accordingly.

Reply:
Done.